packages/unity_video_player/unity_video_player_media_kit/lib/unity_video_player_media_kit.dart

```diff
@@ -0,0 +1,173 @@
+    required this.fit,
+    required this.color,
+  }) : super(key: key);
+
+  final Player player;
+  final BoxFit fit;
+  final Color color;
+
+  @override
+  State<_MKVideo> createState() => __MKVideoState();
+}
+
+class __MKVideoState extends State<_MKVideo> {
+  // Reference to the [VideoController] instance from `package:media_kit_core_video`.
+  VideoController? controller;
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      // Create a [VideoController] instance from `package:media_kit_core_video`.
+      // Pass the [handle] of the [Player] from `package:media_kit` to the [VideoController] constructor.
+      controller = await VideoController.create(widget.player.handle);
+      setState(() {});
+    });
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    if (!widget.player.state.isPlaying) widget.player.play();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Video(
+      controller: controller,
+      fill: widget.color,
+      fit: widget.fit,
+    );
+  }
+}
+
+class UnityVideoPlayerMediaKit extends UnityVideoPlayer {
+  Player mkPlayer = Player();
+
+  @override
+  String? get dataSource =>
+      mkPlayer.state.playlist.medias[mkPlayer.state.playlist.index].uri;
+
+  @override
+  String? get error {
+    // if (mkPlayer.error.isEmpty) return null;
+    // return mkPlayer.error;
+
+    return null;
+  }
+
+  @override
+  Duration get duration => mkPlayer.state.duration;
+
+  @override
+  Duration get currentPos => mkPlayer.state.position;
+
+  @override
+  bool get isBuffering => mkPlayer.state.isBuffering;
+
+  @override
+  bool get isSeekable => true;
+
+  @override
+  Stream<Duration> get onCurrentPosUpdate => mkPlayer.streams.position;
+  @override
+  Stream<bool> get onBufferStateUpdate => mkPlayer.streams.isBuffering;
+
+  @override
+  bool get isPlaying => mkPlayer.state.isPlaying;
+
+  @override
+  Stream<bool> get onPlayingStateUpdate => mkPlayer.streams.isPlaying;
+
+  @override
+  Future<void> setDataSource(String url, {bool autoPlay = true}) async {
+    // do not use mkPlayer.add because it doesn't support auto play
+    mkPlayer.open(Playlist([Media(url)]), play: autoPlay);
+  }
+
+  @override
+  Future<void> setVolume(double volume) async => mkPlayer.volume = volume;
+
+  @override
+  Future<double> get volume => mkPlayer.streams.volume.last;
+
+  @override
+  Future<void> setSpeed(double speed) async => mkPlayer.rate = speed;
+  @override
+  Future<void> seekTo(Duration position) async => mkPlayer.seek(position);
+
+  @override
+  Future<void> start() async => mkPlayer.play();
+  @override
+  Future<void> pause() async => mkPlayer.pause();
+  @override
+  Future<void> release() async {}
+  @override
+  Future<void> reset() async {
+    await pause();
+    await seekTo(Duration.zero);
+  }
+
+  @override
+  void dispose() {
+    mkPlayer.dispose();
```

Comment:
```suggestion
    Future.microtask(() async {
      await mkPlayer.dispose();
      await controller?.dispose();
    });
```

`VideoController` seems to be leaking.